server/events/vcs/github_client.go

```diff
@@ -344,28 +338,22 @@ func (g *GithubClient) GetCombinedStatusMinusApply(repo models.Repo, pull *githu
 	//iterate over check completed check suites - return false if we find one that doesnt have conclusion = "success"
 	for _, c := range checksuites.CheckSuites {
 		if *c.Status == "completed" {
-			fmt.Printf("Looking at suite %v\n", *c.ID)
 			//iterate over the runs inside the suite
 			suite, _, err := g.client.Checks.ListCheckRunsCheckSuite(context.Background(), repo.Owner, repo.Name, *c.ID, nil)
 			if err != nil {
 				return false, errors.Wrap(err, "getting check runs for check suite")
 			}
 
 			for _, r := range suite.CheckRuns {
-				fmt.Printf("Looking at check run %s\n", *r.Name)
 				//check to see if the check is required
 				if isRequiredCheck(*r.Name, required.RequiredStatusChecks.Contexts) {
```

Comment:
This code will crash if no required checks are configured for the repo. While feature flag kind of implies that at least one required check should be set - `atlantis/apply` - it's completely plausible to have 1 or 2 repos in organization misconfigured or deliberately configured in a different way. If there are no required checks it means that we don't even  need to go through CheckRuns. 

I understand it's out of scope of this PR, but feels like the best place to point it out :)

Reply:
I'm not sure what you mean @stasostrovskyi - I just tested this again with no required checks configured and it works fine? It should just pass right through that section if none of the checks are required?

Reply:
This is what we got when running without required checks 
```
Error: goroutine panic. This is a bug.

runtime error: invalid memory address or nil pointer dereference
runtime/panic.go:260 (0x44c9d5)
runtime/signal_unix.go:835 (0x44c9a5)
github.com/runatlantis/atlantis/server/events/vcs/github_client.go:357 (0xa89acc)
github.com/runatlantis/atlantis/server/events/vcs/github_client.go:430 (0xa8a197)
github.com/runatlantis/atlantis/server/events/vcs/instrumented_client.go:179 (0xa90beb)
github.com/runatlantis/atlantis/server/events/vcs/proxy.go:72 (0xa925e4)
github.com/runatlantis/atlantis/server/events/vcs/pull_status_fetcher.go:28 (0xa92e65)
github.com/runatlantis/atlantis/server/events/apply_command_runner.go:109 (0xc72835)
github.com/runatlantis/atlantis/server/events/command_runner.go:296 (0xc77443)
runtime/asm_amd64.s:1594 (0x467ce0)
```

I haven't debugged it but it's either `required` or `RequiredStatusChecks` are `nil` in the case when there are no required checks.

Reply:
GitHub probably returns `nil` ("null" in JSON) instead of an empty list as the code expects.